lib/graphql/analysis/ast/visitor.rb

```diff
@@ -60,6 +60,11 @@ def response_path
           @response_path.dup
         end
 
+        # @return String The a stringified path to the response key for the current field
+        def response_path_str(separator = '.')
+          @response_path.join(separator)
+        end
```

Comment:
What's the rationale for this addition? Should it be added to the child class (`AST::QueryComplexity`) instead, since it's the only one that needs it?

Reply:
The `dup` in `Visitor#response_path` definitely makes sense from a safety perspective but results in an unfortunate array allocation when a client is just going to call `join('.')` on the result. This part of the optimization was a lot more impactful before I isolated the calling of `QueryComplexity#selection_key` to just abstract types. For my test case this part of the change now only saves 1,096 bytes so I don't think it's justified. I'll go ahead and roll this part back.